backends/simplego/exec_special_ops.go

```diff
@@ -850,7 +852,20 @@ func execGatherGeneric[T PODIntegerConstraints](params ...any) any {
 		// Find operand indices:
 		for ii, axis := range startIndexMap {
 			startIndexForAxis := startIndicesFlat[indirectStartIndices[ii]]
-			operandStartIndices[axis] = int(startIndexForAxis)
+			idx := int(startIndexForAxis)
+			// Clamp indices to valid range [0, dim-sliceSize] to match XLA/StableHLO semantics.
+			dim := operandDimensions[axis]
+			maxIdx := dim - sliceSizes[axis]
+			if maxIdx < 0 {
+				// Should not happen if shape inference is correct, but safety first.
+				maxIdx = 0
+			}
+			if idx < 0 {
+				idx = 0
+			} else if idx > maxIdx {
+				idx = maxIdx
+			}
```

Comment:
I suspect that it's a matter of style, since the compiler is likely smart enough to generate the same code in the end, but since Go 1.21 one would write:

```
maxIdx = max(0, maxIdx)
idx = max(0, min(maxIdx, idx))
```

Wdyt?